packages/core/src/utilities/imageRetrieveMetadataProvider.ts

```diff
@@ -0,0 +1,51 @@
+import { addProvider } from '../metaData';
+
+const retrieveOptionsState = new Map<string, any>();
+const retrieveConfigurationState = new Map<string, any>();
+
+const IMAGE_RETRIEVE_CONFIGURATION = 'imageRetrieveConfiguration';
+const IMAGE_RETRIEVE_OPTIONS = 'imageRetrieveOptions';
+/**
+ * Simple metadataProvider object to store metadata for the image retrieval.
+ */
+const imageRetrieveMetadataProvider = {
+  IMAGE_RETRIEVE_CONFIGURATION,
+  IMAGE_RETRIEVE_OPTIONS,
+
+  /** Empty the metadata state */
+  clear: () => {
+    retrieveConfigurationState.clear();
+    retrieveOptionsState.clear();
+  },
+
+  /* Adding a new entry to the state object. */
+  addImageRetrieveConfiguration: (uidOrImageId: string, payload): void => {
+    retrieveConfigurationState.set(uidOrImageId, payload);
+  },
+
+  addImageRetrieveOptions: (uidOrImageId: string, payload): void => {
+    retrieveOptionsState.set(uidOrImageId, payload);
+  },
```

Comment:
this should follow the signiture of other providers which is only an `add` and a `get`